models/iaf_wang_2002_exact.h

```diff
@@ -0,0 +1,516 @@
+    // -----------------------------------------------------------------------
+
+    gsl_odeiv_step* s_;    //!< stepping function
+    gsl_odeiv_control* c_; //!< adaptive stepsize control function
+    gsl_odeiv_evolve* e_;  //!< evolution function
+    gsl_odeiv_system sys_; //!< struct describing system
+
+    /*
+     * integration_step_ should be reset with the neuron on ResetNetwork,
+     * but remain unchanged during calibration. Since it is initialized with
+     * step_, and the resolution cannot change after nodes have been created,
+     * it is safe to place both here.
+     */
+    double step_;             //!< step size in ms
+    double integration_step_; //!< current integration time step, updated by GSL
+
+    /**
+     * Input current injected by CurrentEvent.
+     * This variable is used to transport the current applied into the
+     * _dynamics function computing the derivative of the state vector.
+     * It must be a part of Buffers_, since it is initialized once before
+     * the first simulation, but not modified before later Simulate calls.
+     */
+    double I_stim_;
+  };
+
+
+  // Variables class -------------------------------------------------------
+
+  /**
+   * Internal variables of the model.
+   * Variables are re-initialized upon each call to Simulate.
+   */
+  struct Variables_
+  {
+    //! refractory time in steps
+    long RefractoryCounts;
+  };
+
+  // Access functions for UniversalDataLogger -------------------------------
+
+  //! Read out state vector elements, used by UniversalDataLogger
+  template < State_::StateVecElems elem >
+  double
+  get_ode_state_elem_() const
+  {
+    return S_.ode_state_[ elem ];
+  }
+
+  //! Get NMDA current from state, used by UniversalDataLogger
+  double
+  get_I_NMDA_() const
+  {
+    return S_.I_NMDA_;
+  }
+
+  // Data members -----------------------------------------------------------
+
+  // keep the order of these lines, seems to give best performance
+  Parameters_ P_; //!< Free parameters.
+  State_ S_;      //!< Dynamic state.
+  Variables_ V_;  //!< Internal Variables
+  Buffers_ B_;    //!< Buffers.
+
+  //! Mapping of recordables names to access functions
+  static RecordablesMap< iaf_wang_2002_exact > recordablesMap_;
+
+
+}; /* neuron iaf_wang_2002_exact */
+
+inline size_t
+iaf_wang_2002_exact::send_test_event( Node& target, size_t receptor_type, synindex, bool )
+{
+  SpikeEvent e;
+  e.set_sender( *this );
+  return target.handles_test_event( e, receptor_type );
+}
+
+inline size_t
+iaf_wang_2002_exact::handles_test_event( SpikeEvent&, size_t receptor_type )
+{
+  if ( not( INF_SPIKE_RECEPTOR < receptor_type and receptor_type < SUP_SPIKE_RECEPTOR ) )
+  {
+    throw UnknownReceptorType( receptor_type, get_name() );
+    return 0;
+  }
+  else
+  {
+    if ( receptor_type == SynapseTypes::NMDA )
+    {
+      // give each NMDA synapse a unique rport, starting from 3 (num_ports_ is initialized to 3)
+      ++S_.num_ports_;
```

Comment:
After we have run `init_buffers_()`, i.e., once `B_.e_ != nullptr`, the size of the state vector is fixed and we cannot add any new NMDA connections. We need a check for that here.

Reply:
I added a check for this, see the new version